presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/RowExpressionVerifier.java

```diff
@@ -178,9 +178,6 @@ protected Boolean visitCast(Cast expected, RowExpression actual)
             }
             return getValueFromLiteral(literal).equals(String.valueOf(LiteralInterpreter.evaluate(TEST_SESSION.toConnectorSession(), (ConstantExpression) actual)));
         }
-        if (actual instanceof VariableReferenceExpression && expected.getExpression() instanceof SymbolReference && expected.getType().equals(actual.getType().toString())) {
-            return visitSymbolReference((SymbolReference) expected.getExpression(), actual);
-        }
```

Comment:
This one will consider VariableReferenceExpression "expr" (type varchar) and RowExpression "cast(expr to varchar)" as the same expression, and fails the test in "TestCrossJoinWithOrFilterToInnerJoin.java". While "cast(expr to varchar)" will evaluate to "expr", still we cannot consider them as the same expression. Hence remove here, and it does not break existing tests.